sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,322 @@ use opentitanlib::dif::lc_ctrl::{
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
+    // disconnect us from the LC TAP.
+    tap_strapping
+        .apply()
+        .context("failed to apply PINMUX_TAP_LC strapping")?;
+
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset target")?;
+
+    // Remove the RMA_BOOTSTRAP strapping to bring the device out of the RMA spin loop.
+    rma_bootstrap_strapping
+        .remove()
+        .context("failed to remove strapping RMA_BOOTSTRAP")?;
+
+    // NOTE: without this sleep, the transition starts but isn't successful.
+    // The sleep must happen anywhere before the `START` `TRANSITION_CMD` is given.
+    // Not sure why at the moment, seems like the chip isn't ready even though the
+    // `STATUS.READY` bit is checked.
+    std::thread::sleep(Duration::from_secs(1));
+
+    log::info!("Connecting to JTAG interface");
+
+    let jtag = transport.jtag(&opts.jtag).context("failed to get JTAG")?;
+    jtag.connect(JtagTap::LcTap)
+        .context("failed to connect to JTAG")?;
+
+    // Wait for the lifecycle controller to enter the `READY` state from
+    // which it accepts transition commands.
+    wait_until_ready(&jtag, Duration::from_secs(3))
+        .context("failed to wait for lifecycle controller to be ready")?;
+
+    // Check we're in the `PROD` state with 5 transitions registered.
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?,
+        DifLcCtrlState::Prod.redundant_encoding()
+    );
+    assert_eq!(jtag.read_lc_ctrl_reg(&LcCtrlReg::LcTransitionCnt)?, 5);
+
+    // Attempt to claim the transition mutex.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::ClaimTransitionIf,
+        u8::from(MultiBitBool8::True) as u32,
+    )?;
+    // Will be true if the transition was successful.
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::ClaimTransitionIf)?,
+        u8::from(MultiBitBool8::True) as u32
+    );
+    // And hardware will enable writing to transition registers.
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionRegwen)?,
+        LcCtrlTransitionRegwenBit::TransitionRegwen.mask(),
+    );
+
+    // Multi-reg registers for the transition token.
+    let token_regs = [
+        &LcCtrlReg::TransitionToken0,
+        &LcCtrlReg::TransitionToken1,
+        &LcCtrlReg::TransitionToken2,
+        &LcCtrlReg::TransitionToken3,
+    ];
     // This is the preimage of the RMA token. The postimage is already stored in
     // the OTP's SECRET2 partition.
+    // FIXME: where did this data come from?
     let token = DifLcCtrlToken::from([
         0x53, 0xa3, 0x81, 0x2b, 0x5a, 0x4c, 0x04, 0xa4, //
         0x85, 0xda, 0xac, 0x25, 0x2d, 0x14, 0x5c, 0xaf,
     ]);
```

Comment:
Sorry, this could have been clearer.  I randomly generated the preimage and hardcoded it both here and in [gen_rma_token.py](https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/tests/rom/e2e_bootstrap_rma/gen_rma_token.py;l=8-9;drc=31a5c9c9ad75ce89c6fa67cb5b6237722472b2bb). That python script produces the postimage, which is embedded in the OTP: [sw/device/silicon_creator/rom/e2e/BUILD](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/e2e/BUILD;l=464-475;drc=fa5e362cc3fb703c1993b414c42259fa8c7eb3ad).

Reply:
Thanks @dmcardle, that makes sense